#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/road/protos/RoadIntersection.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/road/protos/StraightRoadSegment.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Pavement.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/vehicles/protos/bmw/BmwX5.proto"

WorldInfo {
  window "<generic>"
  ERP 0.6
  basicTimeStep 10
}
Viewpoint {
  orientation 0.18096050942608105 -0.5610855041183916 -0.8077353224271324 5.523385722051044
  position -50.36795153913075 -46.48382053117258 40.6661639606391
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 4.74 0 -0.01
  floorSize 1000 1000
}
DEF Road_0 StraightRoadSegment {
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_1 StraightRoadSegment {
  translation 23.5 -23.5 0
  rotation 0 0 1 1.5708
  name "road(3)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_2 StraightRoadSegment {
  translation 23.5 55.5 0
  rotation 0 0 1 1.5708
  name "road(10)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_3 StraightRoadSegment {
  translation 55.5 55.5 0
  rotation 0 0 1 1.5708
  name "road(11)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_4 StraightRoadSegment {
  translation 55.5 -23.5 0
  rotation 0 0 1 1.5708
  name "road(9)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_5 StraightRoadSegment {
  translation 23.5 8.5 0
  rotation 0 0 1 1.5708
  name "road(4)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_6 StraightRoadSegment {
  translation 55.5 8.5 0
  rotation 0 0 1 1.5708
  name "road(7)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_7 StraightRoadSegment {
  translation 55.5 23.5 0
  rotation 0 0 1 1.5708
  name "road(8)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_8 StraightRoadSegment {
  translation 23.5 23.5 0
  rotation 0 0 1 1.5708
  name "road(5)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_9 StraightRoadSegment {
  translation 32 0 0
  name "road(1)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_10 StraightRoadSegment {
  translation 32 47 0
  name "road(6)"
  numberOfLanes 1
  length 20
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_11 StraightRoadSegment {
  translation 64 0 0
  name "road(2)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_12 StraightRoadSegment {
  translation 64 47 0
  name "road(13)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_13 StraightRoadSegment {
  translation 0 47 0
  name "road(12)"
  numberOfLanes 1
  length 15
  pavementAppearance Pavement {
    colorOverride 0.666667 0 0.498039
  }
  locked FALSE
  roadBoundingObject TRUE
}
DEF Road_14 Solid {
  translation 23.5 0 0
  children [
    Solid {
      translation -3.89768 -6.01198 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(7)"
    }
    Solid {
      translation -6 -3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(6)"
    }
    Solid {
      translation 3.88229 -6.00815 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(5)"
    }
    Solid {
      translation 6 -3.91 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(4)"
    }
    Solid {
      translation -6.02 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(3)"
    }
    Solid {
      translation -3.89815 5.99803 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(2)"
    }
    Solid {
      translation 3.89186 6.00185 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(1)"
    }
    Solid {
      translation 6 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
    }
    RoadIntersection {
      rotation 0 0 1 -0.7854
      startRoadsNumberOfLanes 1
      startRoadBorder FALSE
      startRoadBorderboundingObject TRUE
      boundingObject TRUE
    }
  ]
  name "CustomRoadIntersection"
}
DEF Road_15 Solid {
  translation 55.5 0 0
  children [
    Solid {
      translation -3.89768 -6.01198 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(7)"
    }
    Solid {
      translation -6 -3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(6)"
    }
    Solid {
      translation 3.88229 -6.00815 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(5)"
    }
    Solid {
      translation 6 -3.91 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(4)"
    }
    Solid {
      translation -6.02 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(3)"
    }
    Solid {
      translation -3.89815 5.99803 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(2)"
    }
    Solid {
      translation 3.89186 6.00185 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(1)"
    }
    Solid {
      translation 6 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
    }
    RoadIntersection {
      rotation 0 0 1 -0.7854
      startRoadsNumberOfLanes 1
      startRoadBorder FALSE
      startRoadBorderboundingObject TRUE
      boundingObject TRUE
    }
  ]
  name "CustomRoadIntersection(1)"
}
DEF Road_16 Solid {
  translation 23.5 47 0
  children [
    Solid {
      translation -3.89768 -6.01198 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(7)"
    }
    Solid {
      translation -6 -3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(6)"
    }
    Solid {
      translation 3.88229 -6.00815 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(5)"
    }
    Solid {
      translation 6 -3.91 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(4)"
    }
    Solid {
      translation -6.02 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(3)"
    }
    Solid {
      translation -3.89815 5.99803 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(2)"
    }
    Solid {
      translation 3.89186 6.00185 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(1)"
    }
    Solid {
      translation 6 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
    }
    RoadIntersection {
      rotation 0 0 1 -0.7854
      startRoadsNumberOfLanes 1
      startRoadBorder FALSE
      startRoadBorderboundingObject TRUE
      boundingObject TRUE
    }
  ]
  name "CustomRoadIntersection(2)"
}
DEF Road_17 Solid {
  translation 55.5 47 0
  children [
    Solid {
      translation -3.89768 -6.01198 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(7)"
    }
    Solid {
      translation -6 -3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(6)"
    }
    Solid {
      translation 3.88229 -6.00815 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(5)"
    }
    Solid {
      translation 6 -3.91 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(4)"
    }
    Solid {
      translation -6.02 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(3)"
    }
    Solid {
      translation -3.89815 5.99803 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(2)"
    }
    Solid {
      translation 3.89186 6.00185 0
      rotation 0 0 1 1.57
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
      name "solid(1)"
    }
    Solid {
      translation 6 3.9 0
      children [
        Shape {
          appearance Pavement {
            colorOverride 0.666667 0 0.498039
          }
          geometry Box {
            size 5 0.8 0.3
          }
        }
      ]
    }
    RoadIntersection {
      rotation 0 0 1 -0.7854
      startRoadsNumberOfLanes 1
      startRoadBorder FALSE
      startRoadBorderboundingObject TRUE
      boundingObject TRUE
    }
  ]
  name "CustomRoadIntersection(3)"
}
DEF greenStart Solid {
  translation 0 47 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.101961 1 0
        metalness 0
      }
      geometry Box {
        size 1 1 10
      }
    }
  ]
  name "greenStart"
}
DEF redEnd Robot {
  translation 23.5 55.5 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.666667 0 0
        metalness 0
      }
      geometry Box {
        size 1 1 10
      }
    }
  ]
  name "redEnd"
  boundingObject Box {
    size 1 1 10
  }
  supervisor TRUE
}
DEF car BmwX5 {
  hidden position_0_0 -6.940127904861203e-05
  hidden position2_0_0 0.0025234878382057123
  hidden position_0_1 -0.00010104060546799578
  hidden position2_0_1 0.0010714084849187172
  hidden position_0_2 1.0661333120501757
  hidden position_0_3 0.89575551510613
  hidden rotation_1 0.24095538715976594 -0.9405586668694763 -0.239353077221499 4.649728876292331
  hidden rotation_2 -0.24100300311304157 0.9429696578609916 -0.22962094164953575 4.607426567516831
  hidden rotation_3 0 -1 0 4.704219085355313
  hidden rotation_4 0 -1 0 3.604242214050586
  translation -0.8727880817534354 1.1157522995507705 0.16806210016159356
  rotation 0.005113138892478476 -0.006050368161454012 0.9999686239356599 2.2545754200860957
  name "car"
  controller "<extern>"
  supervisor TRUE
  sensorsSlotTop [
    Camera {
      translation 1 1 1
      children [
        Pose {
          rotation 0 -1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.8 0.8 0.8
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.1
                radius 0.04
              }
            }
          ]
        }
      ]
      fieldOfView 1
      height 36
    }
  ]
}
